### PR TITLE
Repo rework

### PR DESCRIPTION
**Pull request by sungam3r**
- add GitHub Actions: test, publish, label
- add SourceLink
- add dependabot
- remove appveyor and JS tools

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,28 +0,0 @@
-{
```
Comment (joemcbride) [inline .vscode/launch.json:1]: These files help with VSCode, which I use, and which should be checked into the repo and not removed.

Comment (sungam3r) [reply .vscode/launch.json]: reverted

**Comment by sungam3r** [comment]
For some reason approval test does not work. Locally it works.

File: src/GraphQLParser.ApiTests/GraphQLParser.ApiTests.csproj [hunk 2]
```diff
@@ -5,15 +5,21 @@
   </PropertyGroup>
```
Comment (Shane32) [inline src/GraphQLParser.ApiTests/GraphQLParser.ApiTests.csproj:5]:
```suggestion
    <IsPackable>false</IsPackable>
  </PropertyGroup>
```
???

**Comment by sungam3r** [comment]
Shouldly beta0002 -> 0004 did the trick.

File: src/Directory.Build.props [hunk 3]
```diff
@@ -1,14 +1,33 @@
...
+    <Copyright>Copyright 2016-$(CurrentYear) $(Authors) et al. All rights reserved.</Copyright>
+    <PackageLicenseExpression>MIT</PackageLicenseExpression>
+    <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
+    <PackageIcon>logo.64x64.png</PackageIcon>
+    <RepositoryType>git</RepositoryType>
+    <DebugType>embedded</DebugType>
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+    <!-- https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables -->
+    <ContinuousIntegrationBuild Condition="'$(GITHUB_ACTIONS)' == 'true'">True</ContinuousIntegrationBuild>
```
Comment (sungam3r) [inline src/Directory.Build.props:20]: Thanks, @Shane32

**Review by Shane32 — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-preview.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+      - name: Use .NET Core 3.1 SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
+          source-url: https://nuget.pkg.github.com/sungam3r/index.json
+        # Error: The NUGET_AUTH_TOKEN environment variable was not provided. In this step, add the following:
```
[the remark this answers is not in the document]
Comment (Shane32) [reply .github/workflows/publish-preview.yml:25]: ???

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+          DOTNET_CLI_TELEMETRY_OPTOUT: true
+      - name: Build solution [Release]
+        working-directory: src
+        run: dotnet build --no-restore -c Release
+      - name: Build solution [Debug]
+        working-directory: src
+        run: dotnet build --no-restore -c Debug
+      - name: Test solution [Debug]
+        working-directory: src
+        run: dotnet test --no-restore
```
[the remark this answers is not in the document]
Comment (Shane32) [reply .github/workflows/test.yml:40]: Missing CR/LF from last line in file

File: .github/workflows/publish-preview.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (sungam3r) [reply .github/workflows/publish-preview.yml:25]: Copy-paste

File: src/GraphQLParser.ApiTests/GraphQLParser.ApiTests.csproj (hunk 2 above)
Comment (Shane32) [reply src/GraphQLParser.ApiTests/GraphQLParser.ApiTests.csproj:5]: All csproj files should have `<IsPackable>false</IsPackable>` if it's not going to output nupkg files -- I think.

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      - develop
+    paths: 
+      - src/**
+      - .github/workflows/**
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
[the remark this answers is not in the document]
Comment (sungam3r) [reply .github/workflows/test.yml:16]: Why is it better?

File: .github/workflows/publish-release.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        run: |
+          version="${github_ref:10}"
+          echo version=$version
+          echo "version=$version" >> $GITHUB_ENV
+      - name: Use .NET Core 3.1 SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
+          source-url: https://api.nuget.org/v3/index.json
+        # Error: The NUGET_AUTH_TOKEN environment variable was not provided. In this step, add the following:
```
[the remark this answers is not in the document]
Comment (Shane32) [reply .github/workflows/publish-release.yml]: I'm not sure why we are adding a comment for an error that occurs when you don't follow the instructions...  Of course it's not likely to work.

File: .github/workflows/test.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (Shane32) [reply .github/workflows/test.yml:16]: I suggest this form for lists.  No big deal either way.

**Comment by Shane32** [comment]
Overall 👍

File: .github/workflows/publish-preview.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Shane32) [reply .github/workflows/publish-preview.yml]: The URL should be verified.

File: src/GraphQLParser.ApiTests/GraphQLParser.ApiTests.csproj (hunk 2 above)
Comment (Shane32) [reply src/GraphQLParser.ApiTests/GraphQLParser.ApiTests.csproj:5]: (Also needed in `GraphQLParser.Tests.csproj`)

File: .github/workflows/test.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (Shane32) [reply .github/workflows/test.yml:16]: If you want to leave it, just resolve the conversation.

File: .github/workflows/test.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (sungam3r) [reply .github/workflows/test.yml]: изображение

I did commit from local client.

File: src/GraphQLParser.ApiTests/GraphQLParser.ApiTests.csproj (hunk 2 above)
Comment (sungam3r) [reply src/GraphQLParser.ApiTests/GraphQLParser.ApiTests.csproj:5]: `IsPackable` is always false for test projects so we need to set it explicitly only for a few projects like benchmarks.

File: .github/workflows/publish-preview.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (sungam3r) [reply .github/workflows/publish-preview.yml]: fixed

File: .github/workflows/publish-release.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (sungam3r) [reply .github/workflows/publish-release.yml]: It was copy-paste. OK.

**Comment by sungam3r** [comment]
Is everything OK now?

**Review by Shane32 — APPROVED** [review]

**Pull request merged**